diff --git a/ckan/logic/action/update.py b/ckan/logic/action/update.py
index 7ca6a4ee2..9e2dd3066 100644
--- a/ckan/logic/action/update.py
+++ b/ckan/logic/action/update.py
@@ -251,7 +251,9 @@ def package_update(context, data_dict):
 
     # immutable fields
     data_dict["id"] = pkg.id
-    data_dict['type'] = pkg.type
+    # Allow setting type using context
+    # We don't use type from data_dict for security reasons
+    data_dict["type"] = context.get('type') or pkg.type
 
     _check_access('package_update', context, data_dict)
 
