diff --git a/ckan/authz.py b/ckan/authz.py
index 003dff3b0..97eef415b 100644
--- a/ckan/authz.py
+++ b/ckan/authz.py
@@ -385,13 +385,7 @@ def has_user_permission_for_some_org(user_name, permission):
 def get_user_id_for_username(user_name, allow_none=False):
     ''' Helper function to get user id '''
     # first check if we have the user object already and get from there
-    try:
-        if c.userobj and c.userobj.name == user_name:
-            return c.userobj.id
-    except TypeError:
-        # c is not available
-        pass
-    user = model.User.get(user_name)
+    user = _get_user(user_name)
     if user:
         return user.id
     if allow_none:
