diff --git a/ckan/logic/action/update.py b/ckan/logic/action/update.py
index 70b575eea..ae7f19eb9 100644
--- a/ckan/logic/action/update.py
+++ b/ckan/logic/action/update.py
@@ -651,7 +653,7 @@ def user_update(context, data_dict):
 
     '''
     model = context['model']
-    user = context['user']
+    user = author = context['user']
     session = context['session']
     schema = context.get('schema') or schema_.default_update_user_schema()
     id = _get_or_bust(data_dict, 'id')
@@ -681,7 +683,7 @@ def user_update(context, data_dict):
             }
     activity_create_context = {
         'model': model,
-        'user': user,
+        'user': author,
         'defer_commit': True,
         'ignore_auth': True,
         'session': session
