diff --git a/ckan/logic/action/update.py b/ckan/logic/action/update.py
index 7ca6a4ee2..d42e0430c 100644
--- a/ckan/logic/action/update.py
+++ b/ckan/logic/action/update.py
@@ -932,7 +932,6 @@ def task_status_update(context, data_dict):
     session = model.Session
     context['session'] = session
 
-    user = context['user']
     id = data_dict.get("id")
     schema = context.get('schema') or schema_.default_task_status_schema()
 
